Cesium3DTilesContent/src/BatchTableToGltfStructuralMetadata.cpp

```diff
@@ -791,6 +816,7 @@ void updateExtensionWithJsonScalarProperty(
 
   for (int64_t i = 0; i < propertyTable.count; ++i, ++p, ++it) {
     if (it->IsNull()) {
+      CESIUM_ASSERT(noDataValue.has_value());
```

Comment:
I think the intention is to guarantee that `*noDataValue` is a valid value before dereferencing the `optional`. And it's been a while since I've looked at this code, so I wasn't 100% sure that that will be the case. But after a lookover I'm fairly confident this won't happen.

TL;DR if `noDataValue` were `nullopt`, this would be a developer error, and I don't think it hurts to keep the assert.

Rationale below if you would like it....
<hr>

First we have to execute `findCompatibleTypes` to narrow down how the property could be represented. For scalar properties, we try to settle on a reasonable sentinel value (-1 or 0) if it contains `null` entries. At the end, we verify whether any numeric sentinel values are left. And if there are none, we actually default the type to `STRING` so that we can use the empty string sentinel.

https://github.com/CesiumGS/cesium-native/blob/80cb0d92c6b1e29a7cfa000318fb00f7762ccf41/Cesium3DTilesContent/src/BatchTableToGltfStructuralMetadata.cpp#L654-L659

Looking at where `noData` is set.... the scalar property _should_ have a sentinel value to fall back on. Because otherwise, it would have been turned into a string property. But this is definitely not obvious from a glance.

https://github.com/CesiumGS/cesium-native/blob/80cb0d92c6b1e29a7cfa000318fb00f7762ccf41/Cesium3DTilesContent/src/BatchTableToGltfStructuralMetadata.cpp#L1420-L1436

And in the way we handle JSON string property, we do check whether the sentinel value is present. We don't attempt to write the sentinel value if `noDataValue` is `std::nullopt`. I think we just give up and try to write `null` to the output. 😆 

https://github.com/CesiumGS/cesium-native/blob/80cb0d92c6b1e29a7cfa000318fb00f7762ccf41/Cesium3DTilesContent/src/BatchTableToGltfStructuralMetadata.cpp#L698-L716